src/Angular.js

```diff
@@ -1772,6 +1772,48 @@ function getTestability(rootElement) {
   return injector.get('$$testability');
 }
 
+function rootElementProviderFactory(rootElement) {
+  return ['$shutdownProvider', function($shutdownProvider) {
+    $shutdownProvider.register(function() {
+      if (rootElement.dealoc) {
```

Comment:
oh, this is just proof of how old this PR is, this was removed with 9c5b407fd1e296dd525c129743f2b2b47da4dc0d

WDYT about changing this to

```js
jqLite(rootElement[0].querySelectorAll('*')).removeData();
rootElement.removeData();
```

?